docs/reference/contrib/settings.md

```diff
@@ -0,0 +1,379 @@
+@register_setting
+class GenericSocialMediaSettings(BaseGenericSetting):
+    facebook = models.URLField()
+
+@register_setting
+class SiteSpecificSocialMediaSettings(BaseSiteSetting):
+    facebook = models.URLField()
+```
+
+Links to your settings will appear in the Wagtail admin 'Settings' menu.
+
+## Edit handlers
+
+Settings use edit handlers much like the rest of Wagtail.
+Add a `panels` setting to your model defining all the edit handlers required:
+
+```python
+@register_setting
+class GenericImportantPages(BaseGenericSetting):
+    donate_page = models.ForeignKey(
+        'wagtailcore.Page', null=True, on_delete=models.SET_NULL, related_name='+'
+    )
+    sign_up_page = models.ForeignKey(
+        'wagtailcore.Page', null=True, on_delete=models.SET_NULL, related_name='+'
+    )
+
+    panels = [
+        FieldPanel('donate_page'),
+        FieldPanel('sign_up_page'),
+    ]
+
+@register_setting
+class SiteSpecificImportantPages(BaseSiteSetting):
+    donate_page = models.ForeignKey(
+        'wagtailcore.Page', null=True, on_delete=models.SET_NULL, related_name='+'
+    )
+    sign_up_page = models.ForeignKey(
+        'wagtailcore.Page', null=True, on_delete=models.SET_NULL, related_name='+'
+    )
+
+    panels = [
+        FieldPanel('donate_page'),
+        FieldPanel('sign_up_page'),
+    ]
+```
+
+You can also customize the editor handlers [like you would do for Page model](/advanced_topics/customisation/page_editing_interface.html#customising-the-tabbed-interface) with a custom `edit_handler` attribute:
+
+```python
+from wagtail.admin.panels import TabbedInterface, ObjectList
+
+@register_setting
+class MySettings(BaseGenericSetting):
+    # ...
+    first_tab_panels = [
+        FieldPanel('field_1'),
+    ]
+    second_tab_panels = [
+        FieldPanel('field_2'),
+    ]
+
+    edit_handler = TabbedInterface([
+        ObjectList(first_tab_panels, heading='First tab'),
+        ObjectList(second_tab_panels, heading='Second tab'),
+    ])
+```
+
+## Appearance
+
+You can change the label used in the menu by changing the
+`django.db.models.Options.verbose_name` of your model.
+
+You can add an icon to the menu by passing an `icon` argument to the
+`register_setting` decorator:
+
+```python
+@register_setting(icon='placeholder')
+class GenericSocialMediaSettings(BaseGenericSetting):
+    ...
+    class Meta:
+        verbose_name = "Social media settings for all sites"
+
+@register_setting(icon='placeholder')
+class SiteSpecificSocialMediaSettings(BaseSiteSetting):
+    ...
+    class Meta:
+        verbose_name = "Site-specific social media settings"
+```
+
+For a list of all available icons, please see the [styleguide](/contributing//styleguide.html).
+
+## Using the settings
+
+Settings can be used in both Python code and in templates.
+
+### Using in Python
+
+#### Generic settings
+
+If you require access to a generic setting in a view, the
+`wagtail.contrib.settings.models.BaseGenericSetting.get_or_create` method allows
```

Comment:
I feel like we could drop the module path here too

```suggestion
`BaseGenericSetting.get_or_create` method allows
```

Reply:
I find the full path helpful, though happy to take your lead for consistency here!